File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/styling/span/internal/NoExtSpanImpl.java

package io.github.abc15018045126.sora.lang.styling.span.internal;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Objects;

import io.github.abc15018045126.sora.lang.styling.Span;
import io.github.abc15018045126.sora.lang.styling.SpanPool;
import io.github.abc15018045126.sora.lang.styling.color.ResolvableColor;
import io.github.abc15018045126.sora.lang.styling.span.SpanExt;

/**
 * Span without SpanExt support.
 *
 * @author abc15018045126
 */
public class NoExtSpanImpl implements Span {
    private final static SpanPool<NoExtSpanImpl> pool = new SpanPool<>(NoExtSpanImpl::new);

    private int column;
    private long style;
    private Object extra;

    NoExtSpanImpl() {

    }

    NoExtSpanImpl(int column, long style) {
        this.column = column;
        this.style = style;
    }

    public static NoExtSpanImpl obtain(int column, long style) {
        return pool.obtain(column, style);
    }


    @Override
    public void setColumn(int column) {
        this.column = column;
    }

    @Override
    public int getColumn() {
        return this.column;
    }

    @Override
    public void setStyle(long style) {
        this.style = style;
    }

    @Override
    public long getStyle() {
        return this.style;
    }

    @Override
    public void setUnderlineColor(@Nullable ResolvableColor color) {
        throw new UnsupportedOperationException();
    }

    @Nullable
    @Override
    public ResolvableColor getUnderlineColor() {
        return null;
    }

    @Override
    public void setExtra(Object extraData) {
        this.extra = extraData;
    }

    @Override
    @Nullable
    public Object getExtra() {
        return this.extra;
    }

    @Override
    public void setSpanExt(int extType, @Nullable SpanExt ext) {
        throw new UnsupportedOperationException();
    }

    @Override
    public boolean hasSpanExt(int extType) {
        return false;
    }

    @Nullable
    @Override
    public <T> T getSpanExt(int extType) {
        return null;
    }

    @Override
    public void removeAllSpanExt() {

    }

    @Override
    public void reset() {
        setColumn(0);
        setStyle(0L);
        extra = null;
    }

    @NonNull
    @Override
    public Span copy() {
        return new NoExtSpanImpl(this.column, this.style);
    }

    @Override
    public boolean recycle() {
        reset();
        return pool.offer(this);
    }

    @Override
    public String toString() {
        return "NoExtSpanImpl{" +
                "column=" + column +
                ", style=" + style +
                ", extra=" + extra +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        NoExtSpanImpl noExtSpan = (NoExtSpanImpl) o;
        return column == noExtSpan.column && style == noExtSpan.style && Objects.equals(extra, noExtSpan.extra);
    }

    @Override
    public int hashCode() {
        return Objects.hash(column, style, extra);
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/styling/span/internal/NoExtSpanImpl.kt
package io.github.abc15018045126.sora.lang.styling.span.internal

import io.github.abc15018045126.sora.lang.styling.Span
import io.github.abc15018045126.sora.lang.styling.SpanPool
import io.github.abc15018045126.sora.lang.styling.color.ResolvableColor
import io.github.abc15018045126.sora.lang.styling.span.SpanExt
import java.util.*

/**
 * Span without SpanExt support.
 *
 * @author abc15018045126
 */
class NoExtSpanImpl : Span {

    override var column: Int = 0
    override var style: Long = 0L
    override var extra: Any? = null

    internal constructor()

    internal constructor(column: Int, style: Long) {
        this.column = column
        this.style = style
    }

    override var underlineColor: ResolvableColor?
        get() = null
        set(value) {
            throw UnsupportedOperationException()
        }

    override fun setSpanExt(extType: Int, ext: SpanExt?) {
        throw UnsupportedOperationException()
    }

    override fun hasSpanExt(extType: Int): Boolean {
        return false
    }

    override fun <T> getSpanExt(extType: Int): T? {
        return null
    }

    override fun removeAllSpanExt() {
    }

    override fun reset() {
        column = 0
        style = 0L
        extra = null
    }

    override fun copy(): Span {
        val span = NoExtSpanImpl(column, style)
        span.extra = extra
        return span
    }

    override fun recycle(): Boolean {
        reset()
        return pool.offer(this)
    }

    override fun toString(): String {
        return "NoExtSpanImpl(column=$column, style=$style, extra=$extra)"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as NoExtSpanImpl
        return column == that.column && style == that.style && extra == that.extra
    }

    override fun hashCode(): Int {
        return Objects.hash(column, style, extra)
    }

    companion object {
        @JvmStatic
        private val pool = SpanPool { c, s -> NoExtSpanImpl(c, s) }

        @JvmStatic
        fun obtain(column: Int, style: Long): NoExtSpanImpl {
            return pool.obtain(column, style)
        }
    }
}
